src/Microsoft.AspNetCore.Http/Internal/DefaultHttpRequest.cs

```diff
@@ -151,12 +156,19 @@ public override Task<IFormCollection> ReadFormAsync(CancellationToken cancellati
             return FormFeature.ReadFormAsync(cancellationToken);
         }
 
+        public override RouteValueDictionary RouteValues
+        {
+            get => RouteValuesFeature.Values;
+            set => RouteValuesFeature.Values = value;
```

Comment:
Is this the right way to add a feature with `HttpRequest`?

Reply:
Yep. Passing in the HttpRequest isn’t right though.